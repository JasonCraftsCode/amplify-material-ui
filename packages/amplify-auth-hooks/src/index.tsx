export * from './use-auth-context';
export * from './use-auth';
export * from './use-check-contact';
export * from './use-confirm-sign-in';
export * from './use-confirm-sign-up';
export * from './use-forgot-password';
export * from './use-notification-context';
export * from './use-require-new-password';
export * from './use-sign-in';
export * from './use-sign-out';
export * from './use-sign-up';
export * from './use-totp-setup';
export * from './use-verify-contact';
