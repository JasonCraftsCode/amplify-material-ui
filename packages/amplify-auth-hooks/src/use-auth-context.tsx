import * as React from 'react';

import { createNamedContext } from './utils';

export type AuthData = any;

export interface AuthState {
  authState: string;
  authData?: AuthData;
}

export interface AuthContextProps extends AuthState {
  handleStateChange: (authState: string, authData: AuthData) => void;
}

export const AuthContext = createNamedContext<AuthContextProps | null>(
  'Auth',
  null
);

export const useAuthContext = (): AuthContextProps =>
  React.useContext(AuthContext) as AuthContextProps;
