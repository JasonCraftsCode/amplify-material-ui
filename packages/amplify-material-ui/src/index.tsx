import { I18n } from '@aws-amplify/core';

import dict from './dict';

export * from './auth';
export * from './ui';
export * from './i18n';

I18n.putVocabularies(dict);
