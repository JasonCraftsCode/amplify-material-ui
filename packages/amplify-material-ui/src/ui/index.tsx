export { default as FormContaier } from './form-container';
export { default as FormSection } from './form-section';
export { default as SectionBody } from './section-body';
export { default as SectionFooter } from './section-footer';
export { default as SectionHeader } from './section-header';
export * from './toast';
export * from './password-strenght-meter';
