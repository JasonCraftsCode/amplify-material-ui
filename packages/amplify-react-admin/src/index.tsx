export * from 'ra-core';
export * from './admin';
