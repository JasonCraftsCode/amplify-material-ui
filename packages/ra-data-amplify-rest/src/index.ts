import {
    UPDATE_MANY,
    DELETE_MANY,
    UPDATE,
    DELETE,
    LegacyDataProvider,
    convertLegacyDataProvider,
} from 'ra-core';

import { buildDataRequest } from './build-data-request';
import { parseResponse } from './parse-response';
import {
    buildDataProvider,
    BuildDataProviderOptions,
} from './build-data-provider';

import { Params } from './types';

const defaultOptions = {
    buildDataRequest,
    parseResponse,
};

export interface DataProviderOptions extends Partial<BuildDataProviderOptions> {
    apiName: string;
}

export default (options: DataProviderOptions): LegacyDataProvider => {
    const dataProvider = buildDataProvider({ ...defaultOptions, ...options });

    return convertLegacyDataProvider(
        async (
            type: string,
            resource: string,
            params: Params,
        ): Promise<any> => {
            if (type === UPDATE_MANY) {
                const { ids, data, ...otherParams } = params;

                if (!ids) return;

                const responses = await Promise.all(
                    ids.map(id =>
                        dataProvider(UPDATE, resource, {
                            data: {
                                id,
                                ...data,
                            },
                            ...otherParams,
                        }),
                    ),
                );

                return {
                    data: responses.map(response => response.data),
                };
            }

            if (type === DELETE_MANY) {
                const { ids, ...otherParams } = params;

                if (!ids) return;

                const responses = await Promise.all(
                    ids.map(id =>
                        dataProvider(DELETE, resource, {
                            id,
                            ...otherParams,
                        }),
                    ),
                );

                return {
                    data: responses.map(response => response.data),
                };
            }

            return dataProvider(type, resource, params);
        },
    );
};

export { buildDataRequest, parseResponse, buildDataProvider };

export * from './types';
