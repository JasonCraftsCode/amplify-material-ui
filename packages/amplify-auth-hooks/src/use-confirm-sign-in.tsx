import invariant from 'tiny-invariant';
import Auth from '@aws-amplify/auth';
import { ConsoleLogger as Logger } from '@aws-amplify/core';

import { useAuthContext } from './use-auth-context';
import { useNotificationContext } from './use-notification-context';
import { useCheckContact } from './use-check-contact';

const logger = new Logger('useConfirmSignIn');

export const useConfirmSignIn = () => {
  const { authData } = useAuthContext();
  const { showNotification } = useNotificationContext();
  const checkContact = useCheckContact();

  const mfaType =
    authData && authData.challengeName === 'SOFTWARE_TOKEN_MFA'
      ? 'TOTP'
      : 'SMS';

  const confirm = async (code: string): Promise<void> => {
    invariant(
      Auth && typeof Auth.confirmSignIn === 'function',
      'No Auth module found, please ensure @aws-amplify/auth is imported'
    );

    try {
      await Auth.confirmSignIn(
        authData,
        code,
        mfaType === 'TOTP' ? 'SOFTWARE_TOKEN_MFA' : null
      );
      checkContact(authData);
    } catch (error) {
      logger.error(error);
      showNotification({ content: error.message, variant: 'error' });
    }
  };

  return {
    confirm,
    mfaType,
  };
};
