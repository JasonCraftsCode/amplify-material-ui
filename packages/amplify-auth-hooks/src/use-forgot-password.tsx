import * as React from 'react';
import invariant from 'tiny-invariant';
import Auth from '@aws-amplify/auth';
import { ConsoleLogger as Logger } from '@aws-amplify/core';

import { useAuthContext } from './use-auth-context';
import { useNotificationContext } from './use-notification-context';

const logger = new Logger('useForgotPassword');

export const useForgotPassword = () => {
  const [delivery, setDelivery] = React.useState(null);
  const { handleStateChange } = useAuthContext();
  const { showNotification } = useNotificationContext();

  const submit = async (
    code: string,
    username: string,
    password: string
  ): Promise<void> => {
    invariant(
      Auth && typeof Auth.forgotPasswordSubmit === 'function',
      'No Auth module found, please ensure @aws-amplify/auth is imported'
    );

    try {
      await Auth.forgotPasswordSubmit(username, code, password);
      handleStateChange('signIn', null);
      setDelivery(null);
    } catch (error) {
      logger.error(error);
      showNotification({ content: error.message, variant: 'error' });
    }
  };

  const send = async (username: string): Promise<void> => {
    invariant(
      Auth && typeof Auth.forgotPassword === 'function',
      'No Auth module found, please ensure @aws-amplify/auth is imported'
    );

    try {
      const data = await Auth.forgotPassword(username);
      setDelivery(data.CodeDeliveryDetails);
    } catch (error) {
      logger.error(error);
      showNotification({ content: error.message, variant: 'error' });
    }
  };

  return {
    delivery,
    submit,
    send,
  };
};
