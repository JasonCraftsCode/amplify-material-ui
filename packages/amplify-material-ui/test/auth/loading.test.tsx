import * as React from 'react';
import { render } from '@testing-library/react';
import { Loading } from '../../src';

describe('loading', () => {
  it('should be rendered correctly', () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
  });
});
