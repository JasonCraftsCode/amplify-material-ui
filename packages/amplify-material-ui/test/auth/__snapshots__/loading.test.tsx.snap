// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 makeStyles-paper-2 MuiPaper-rounded"
    >
      <div
        class="MuiCircularProgress-root makeStyles-progress-1 MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
